doc/nrf/gs_modifying.rst

```diff
@@ -14,27 +14,53 @@ After programming and testing an application, you probably want to make some mod
 Adding files and changing compiler settings
 *******************************************
 
-All files that your application uses must be specified in the :file:`CMakeLists.txt` file.
-By default, most samples include only the main application file :file:`src/main.c`.
-This means that you must add all other files that you are using.
+The |NCS| build system is based on Zephyr, whose build system is based on `CMake`_.
+For more information, see :ref:`zephyr:application` in the Zephyr documentation.
 
-You can also configure compiler options, application defines, or include directories, or set :ref:`build types <gs_modifying_build_types>` in :file:`CMakeLists.txt`.
+In the |NCS|, the application is a CMake project.
+As such, the application controls the configuration and build process of both itself, Zephyr, and sourced libraries.
+The application's :file:`CMakeLists.txt` file is the main CMake project file and the source of the build process configuration.
+
+Zephyr creates a CMake component that must be imported by the application into its :file:`CMakeLists.txt` file.
+When imported, the application is able to reference items provided by both Zephyr and the |NCS|.
+
+Importing the Zephyr `CMake`_ component creates the ``app`` library.
+All files that your application uses must be added to this library.
+This is done directly in the :file:`CMakeLists.txt` file or files that are sourced by it.
+
+The application has full control over the build process it can impact compiler options, application defines, or include directories.
+It can also define how and which libraries are linked with the ``app`` library using either standard CMake commands or commands provided by Zephyr's CMake component.
+
+.. note::
+   Zephyr provides a :kconfig:`CONFIG_COMPILER_OPT` option to add compiler flags visible during entire build process.
```

Comment:
Is this really something we should mention here ?

Readers might belive this is the right way to go, but Zephyr supports most options through Kconfig, like:
Optimization: https://docs.zephyrproject.org/latest/reference/kconfig/choice_555.html#choice-555
Standards: https://docs.zephyrproject.org/latest/reference/kconfig/CONFIG_STD_CPP98.html
Misra sane flags: https://docs.zephyrproject.org/latest/reference/kconfig/CONFIG_MISRA_SANE.html
Coverage: https://docs.zephyrproject.org/latest/reference/kconfig/CONFIG_COVERAGE.html
Color diagnostic: https://docs.zephyrproject.org/latest/reference/kconfig/CONFIG_COMPILER_COLOR_DIAGNOSTICS.html
Different warning bases / levels using `-DW=[1|2|3]`.
and more.

So I would prefer to direct the reader to the known supported flags, and only if the reader needs specific flags not already covered natively by the build system, we should have a description on how to do that.

Reply:
I added it for the reason you mentioned - Zephyr has it's own ways of influencing the build so not everything should go through the CMake files (even if it's possible). Maybe this section should be more descriptive or point somewhere to Zephyr documentation...

Reply:
at least I think we should point to somewhere which describes how to use Kconfig to control the compile options.

Referring directly to `CONFIG_COMPILER_OPT` bears the risk of leading the reader to sub-optimal flow.

I don't have a good proposal atm, but will do some thinking.